File: DSA/Gautam_Shyamnani_590013078/Day38Question1.cpp
#include <iostream>
#include <vector>
#include <stack>
using namespace std;

vector<int> nextGreaterElements(const vector<int>& nums) {
    int n = nums.size();
    vector<int> result(n, -1);  
    stack<int> s;  

    for (int i = 0; i < n; ++i) {
        
        while (!s.empty() && nums[i] > nums[s.top()]) {
            result[s.top()] = nums[i];  
            s.pop();  
        }
        s.push(i);  
    }

    return result;
}

int main() {
    vector<int> arr = {4, 5, 2, 25};
    vector<int> ans = nextGreaterElements(arr);

    for (int x : ans) {
        cout << x << " ";
    }
    cout << endl;

    return 0;
}

File: DSA/N_PARTH_PANDEY_590012761/Day38Question1.py
arr = list(map(int, input().split()))
stack = []
result = [-1] * len(arr)

for i in range(len(arr)):
    while stack and arr[stack[-1]] < arr[i]:
        result[stack.pop()] = arr[i]
    stack.append(i)

print(result)